openprescribing/matrixstore/cachelib.py

```diff
@@ -0,0 +1,60 @@
+"""
+Provides a basic caching decorator with a few special features:
+
+    * Arguments must all be either one of the basic types defined below or have
+      a `cache_key` attribute which identifies the state of the object for
+      caching purposes. Passing in anything else is an error.
+
+    * There is no time-based expiration: the intention is that the decorator
+      should only be applied to functions whose output is purely determined by
+      their arguments. If the logic of the function changes then the `version`
+      argument can be incremented.
+"""
+import functools
+
+from django.core.cache import cache as default_cache
+
+
+MISSING = object()
+BASIC_TYPES = (bool, int, float, str)
+
+
+def memoize(version=1, cache=default_cache):
+    def decorator(func):
+        cache_key_base = _get_cache_key_base(func, version)
+
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            cache_key = _get_cache_key(cache_key_base, args, kwargs)
+            result = cache.get(cache_key, default=MISSING)
+            if result is MISSING:
+                result = func(*args, **kwargs)
+                cache.set(cache_key, result)
+            return result
+
+        return wrapper
+
+    return decorator
```

Comment:
Apologies if I've missed something in the PR -- are there tests for this/for the diskcache?